core/modules/views_ui/views_ui.admin.inc

```diff
@@ -3769,86 +3769,108 @@ function views_ui_add_item_form($form, &$form_state) {
   // Add the display override dropdown.
   views_ui_standard_display_dropdown($form, $form_state, $section);
 
-  // Figure out all the base tables allowed based upon what the relationships provide.
+  // Figure out base tables allowed based upon what the relationships provide.
   $base_tables = $view->get_base_tables();
-  $options = views_fetch_fields(array_keys($base_tables), $type, $display->handler->use_group_by());
+  $fields = views_fetch_fields(array_keys($base_tables), $type, $display->handler->use_group_by());
 
-  if (!empty($options)) {
-    $form['options']['controls'] = array(
-      '#theme_wrappers' => array('container'),
-      '#id' => 'views-filterable-options-controls',
-      '#attributes' => array('class' => array('container-inline')),
+  $form['options']['controls'] = array(
+    '#theme_wrappers' => array('container'),
+    '#id' => 'views-filterable-options-controls',
+    '#attributes' => array('class' => array('container-inline')),
+  );
+  $form['options']['controls']['options_search'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Search'),
+  );
+
+  $form['options']['controls']['group'] = array(
+    '#type' => 'select',
+    '#title' => t('Filter'),
+    '#options' => array(),
+    '#empty_value' => 'all',
+    '#empty_option' => t('- All -'),
+  );
+
+  $form['options']['name'] = array(
+    '#prefix' => '<div class="views-radio-box form-checkboxes views-filterable-options">',
+    '#suffix' => '</div>',
+    '#tree' => TRUE,
+  );
+
+  // Build the table header.
+  $header = array(
+    'title' => array('data' => t('Title'), 'sort' => 'asc'),
+    'group' => array('data' => t('Category'), 'sort' => 'asc'),
+    'help' => array('data' => t('Description')),
+  );
+  $groups = array();
+  foreach ($fields as $key => $field) {
+    $group = preg_replace('/[^a-z0-9]/', '-', strtolower($field['group']));
+    $groups[$group] = $field['group'];
+    $fields[$key]['#attributes'] = array(
+      'class' => array(
+        'views-filterable-option',
+        $group,
+      ),
     );
-    $form['options']['controls']['options_search'] = array(
-      '#type' => 'textfield',
-      '#title' => t('Search'),
+    $fields[$key]['help'] = array(
+      'data' => $field['help'],
+      'class' => array('description'),
     );
-
-    $groups = array('all' => t('- All -'));
-    $form['options']['controls']['group'] = array(
-      '#type' => 'select',
-      '#title' => t('Filter'),
-      '#options' => array(),
+    // Here we generate what will be the ID of the checkbox for this $key.
+    // This allows us to tie the label to the checkbox for better accessibility.
+    $checkbox_id = 'edit-name-fields-' . backdrop_html_class($key);
```

Comment:
I'm not sure which is best here, the way it is concatenated now, or like this instead:
```suggestion
    $checkbox_id = backdrop_html_class('edit-name-fields-' . $key);
```

In any case, can we please add something to the inline comment to explain why we are intentionally using `backdrop_html_class()` here instead of `backdrop_html_id()`?